src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java

```diff
@@ -539,6 +559,23 @@ public void forecast(Weather weather, long duration) {
         }
     }
 
+    @Inject(method = "updateWeather", at = @At(value = "RETURN"))
+    public void onUpdateWeatherReturn(CallbackInfo ci) {
+        Weather weather = getWeather();
+        if (this.prevWeather != weather) {
```

Comment:
`prevWeather` can be null here- it might be necessary to update the API to make `initalWeather an `Optional`